### DISCUSSION TITLE
OBS

### DISCUSSION THREAD

**Discussion by bittnada** [The-OpenROAD-Project/OpenROAD#2543] [Q&A]
Hi, do you know the role of macro that has only OBS without pin. This question is actually irrelevant to OpenROAD, but I
need a help to understand this.
I found that some of macros, described in the section of "COMPONENTS" in def file, is actually included in none of net.
And I also found that those weird macros have only OBS without PIN in LEF. I am quite new in VLSI filed, so I don't understand the meaning of this kind of macros at all.
I would be glad if any one can help.

Thanks.

**Comment by rovinski** [comment]
There are a few uses for macros without pins. The primary one is for metrology cells: cells that the fab requires you to insert in order to fabricate the chip. These cells can be marker cells which are used to optically align the chip between lithography steps. There can also be cells for measuring the process steps, e.g. whether an etch or lithography step was within the fab's process parameters or not, and to tune equipment.

I can also imagine there being cells which are meant to be contacts (or breakers) to the substrate. Technically, these should have the substrate contacts defined as pins, but never doubt that a PDK has been poorly defined. On that note, don't rule out that the PDK could simply have an error in it and the cell was improperly extracted.

[Here](https://www.mems-exchange.org/users/masks/intro-equipment.html) is a decent reference on alignment markers.

**Reply by bittnada** [reply]
@rovinski 
Thanks a lot. 
can I ask you further question? Then, when are these macros designed to be inserted in PNR ?
Does fabs require to insert those macros before the steps of floorplanning and macro placement?
Thanks.

**Reply by rovinski** [reply]
It depends on the fab and the process. There are 3 main cases I have seen:

1. The most common approach is that the fab provides a cell as a block macro (larger than a standard cell). These cells should be placed during macro placement. The main concern is that these cells usually cannot be automatically placed (unless your fab provides you a script to do so in your implementation tool). The designer will have to be aware of the fab's requirements and place the cell accordingly.
2. The fab provides macro(s) which have the same footprint as a standard cell. In this case, the foundry may provide you with a script to place the cell or you may need to make your own script / place it manually.
3. Some fabs directly integrate these cells into a seal ring, in which case the uses does not have to worry about additional effort to place these cells. The seal ring is merged as one of the last steps in the design.

Additionally, some fabs / processes may not require placement of these cells at all because the fab inserts them elsewhere. Usually if this is the case, however, the fab wouldn't provide these cells in your PDK.

Sometimes the easiest way to check the placement (other than checking a manual or asking a fab representative) is to run a design rule check (DRC). These rules will be encoded in DRC and can often tell you where / how many fiducial cells you need. Other times, the DRC can be overly confusing and may give you improper results if you don't have the correct DRC options turned on 🤷‍♂️. The best way is to always ask a person for clarification.

**Reply by bittnada** [reply]
@rovinski 
I really appreciate for your kind help.
